packages/@sanity/core/src/actions/dataset/alias/validateDatasetAliasName.js

```diff
@@ -0,0 +1,33 @@
+module.exports = (datasetName) => {
+  if (!datasetName) {
+    return 'Alias name is missing'
+  }
+
+  const name = `${datasetName}`
+
+  if (name.toLowerCase() !== name) {
+    return 'Alias name must be all lowercase characters'
+  }
+
+  if (name.length < 2) {
+    return 'Alias name must be at least two characters long'
+  }
+
+  if (name.length > 20) {
+    return 'Alias name must be at most 20 characters'
+  }
+
+  if (!/^[a-z0-9~]/.test(name)) {
+    return 'Alias name must start with a letter or a number'
+  }
+
+  if (!/^[a-z0-9~][-_a-z0-9]+$/.test(name)) {
+    return 'Alias name must only contain letters, numbers, dashes and underscores'
+  }
+
+  if (/[-_]$/.test(name)) {
+    return 'Alias name must not end with a dash or an underscore'
+  }
+
+  return false
+}
```

Comment:
This may be waaaaay late, but should we try to re-use existing validations?

We got one we recently tweaked: `packages/@sanity/client/src/validators.js`.

Reply:
The code already provides better error messages though. Not sure if we should change this now.

Reply:
We shouldn't import from the client internals, so I feel like this is fine.

Reply:
Well I do have objections depending on client package for validations, but either way, you will have to update even datasets too.